deferred_heap.h

```diff
@@ -87,43 +109,38 @@ namespace gcpp {
 			dtors.clear();
 		}
 
-		//	Runn all the destructors for objects in [begin,end)
+		//	Run all the destructors for objects in [begin,end)
 		//
 		bool run(gsl::span<byte> range) {
 			if (range.size() == 0)
 				return false;
 
-			bool ret = false;
-
 			//	for reentrancy safety, we'll take a local copy of destructors to be run
 			//
-			//	first, move any destructors for objects in this range to a local list...
+			//	move any destructors for objects in this range to a local list...
 			//
-			std::vector<destructor> to_destroy;
-			for (auto it = dtors.begin(); it != dtors.end(); /*--*/) {
-				//	<= and -- to avoid dereferencing a past-the-end iterator
-				if (&*range.begin() <= it->p && it->p <= &*--range.end()) {
```

Comment:
Just confirming I parsed that correctly: Do you mean `++`/`--` aren't guaranteed to work on a temporary iterator as in `--range.end()`?

I didn't know that, thanks... just curious why the iterator's value category would matter?